src/stores/DataStore.js

```diff
@@ -46,6 +46,10 @@ class DataStore extends Collection {
     if (typeof idOrInstance === 'string') return idOrInstance;
     return null;
   }
+
+  static get [Symbol.species]() {
+    return Collection;
```

Comment:
`return this`

Reply:
@devsnek That would just return the subclass again, causing the same error as before.

```js
class Test { static get [Symbol.species] () { return this; } }
class Test2 extends Test { }
console.log(Test2[Symbol.species]);
```
Prints out ``[Function: Test2]``

Reply:
yes, that is the intention...

Reply:
Well, in that case, the DataStores and other internal subclasses should be updated to setting the species to Collection to avoid getting the same error as #2710, which PR #2712 introduced. Just returning `this` here without altering other parts of the PR would just introduce the same issue again

Reply:
whoops didn't notice this was in `DataStore.js`

Reply:
... Neither did I, actually :joy: